src/js/gutenberg-plugin.js

```diff
@@ -184,6 +187,217 @@ const ClassifAIGenerateTagsButton = () => {
 	);
 };
 
+let errorCode = '';
+
+/**
+ * Calls the server-side method that synthesizes speech for a post.
+ *
+ * @param {number} postId The Post ID
+ * @return {boolean} Returns true on success.
+ */
+const synthesizeSpeech = async ( postId ) => {
+	const { select, dispatch } = wp.data;
+
+	// Endpoint URL.
+	const synthesizeSpeechUrl = `${ wpApiSettings.root }classifai/v1/synthesize-speech/${ postId }`;
+
+	// Stores status of the synthensis process.
+	const isProcessing = select( postAudioStore ).getIsProcessing();
+
+	// Return early if already processing.
+	if ( isProcessing ) {
+		return;
+	}
+
+	// Set state indicating the synthesis process has begun.
+	dispatch( postAudioStore ).setIsProcessing( true );
+
+	const response = await fetch(
+		synthesizeSpeechUrl,
+		{
+			headers: new Headers( {
+				'X-WP-Nonce': wpApiSettings.nonce,
+			} ),
+		}
+	);
+
+	// Return false if error.
+	if ( 200 !== response.status ) {
+		// Set state indicating the synthesis process has ended.
+		dispatch( postAudioStore ).setIsProcessing( false );
+		return false;
+	}
+
+	const result = await response.json();
+
+	if ( result.success ) {
+		// Set audio ID state after successful synthesis.
+		dispatch( postAudioStore ).setAudioId( result.audio_id );
+
+		// Set state indicating the synthesis process has ended.
+		dispatch( postAudioStore ).setIsProcessing( false );
+
+		if ( errorCode ) {
+			dispatch( noticesStore ).removeNotice( errorCode );
+			errorCode = '';
+		}
+
+		return true;
+	}
+	errorCode = result.code;
+	dispatch( 'core/notices' ).createErrorNotice( result.message, {
+		id: errorCode,
+	} );
+	dispatch( postAudioStore ).setIsProcessing( false );
+};
+
+/**
+ * ClassifAI Text to Audio component.
+ *
+ * @param {Object} props Props object.
+ */
+const ClassifAITSpeechSynthesisToggle = ( props ) => {
+	// State of the audio being previewed in PluginDocumentSettingPanel.
+	const [ isPreviewing, setIsPreviewing ] = useState( false );
+
+	const [ timestamp, setTimestamp ] = useState( new Date().getTime() );
+
+	// Indicates whether speech synthesis is supported for the current post.
+	let isFeatureSupported = false;
+
+	// Indicates whether speech synthesis is enabled for the current post.
+	const isSynthesizeSpeech = 'yes' === useSelect( ( select ) => select( 'core/editor' ).getEditedPostAttribute( 'classifai_synthesize_speech' ) );
+
+	// Post type of the current post.
+	const postType = useSelect( ( select ) => select( 'core/editor' ).getCurrentPostType() );
+
+	// Says whether speech synthesis is in progress.
+	const isProcessingAudio = useSelect( ( select ) => select( postAudioStore ).getIsProcessing() );
+
+	// Figure out if speech synthesis is supported by the current post.
+	if ( classifaiTextToSpeechData && classifaiTextToSpeechData.supportedPostTypes.includes( postType ) ) {
```

Comment:
I'm getting an error here due to this not being defined: `ReferenceError: classifaiTextToSpeechData is not defined`

This crashes the block editor. This happens when I don't have proper authentication in place and as such, the `register` method doesn't fire and `enqueue_editor_assets` never runs

Reply:
Okay I was able to reproduce this after some time. This happens when NLU is also enabled which loads this file which causes it to break.